app/src/main/kotlin/compiler/intermediate_form/FunctionDetailsGenerator.kt

```diff
@@ -1,34 +1,58 @@
 package compiler.intermediate_form
 
+import compiler.ast.Function
+import compiler.ast.Type
 import compiler.ast.Variable
 
-data class FunctionDetailsGenerator(
-    val depth: Int,
-    val vars: Map<Variable, Boolean>,
-    val parameters: List<Variable>
-)
 data class FunctionCallIntermediateForm(
     val callGraph: ControlFlowGraph,
     val result: IntermediateFormTreeNode?
 )
 
-fun generateCall(args: List<IntermediateFormTreeNode>): FunctionCallIntermediateForm {
-    return TODO()
-}
+val FUNCTION_RESULT_REGISTER = Register()
 
-fun genPrologue(): ControlFlowGraph {
-    return TODO()
-}
+data class FunctionDetailsGenerator(
+    val parameters: List<Variable>,
+    val functionCFG: ControlFlowGraph,
+    val function: Function
+) {
+    fun genCall(
+        args: List<IntermediateFormTreeNode>,
+    ): FunctionCallIntermediateForm {
+        val cfgBuilder = ControlFlowGraphBuilder()
 
-fun genEpilogue(): ControlFlowGraph {
-    return TODO()
-}
+        var last: Pair<IFTNode, CFGLinkType>? = null
+        // write arg values to params
+        for ((arg, param) in args zip parameters) {
+            val node = genWrite(param, arg, true)
```

Comment:
`genCall` will be called from the outer function, so this write might not be direct?

Reply:
Changed to `false`, but still not sure this is correct. Should we add `isDirect` also to `genCall`, or is there any method to decide what to put there?